From 3dfcb7afa68e8614ed3362b1b955b72154e892a5 Mon Sep 17 00:00:00 2001
From: Matt Ranostay <matt.ranostay@intel.com>
Date: Tue, 9 Feb 2016 12:59:39 -0800
Subject: [PATCH 4/5] serial: hsu: check for division by zero

baud setting could be zero which could cause an division by zero
fault.

Change-Id: I1ea913cb65a08d0b07a1d1857abe2e2a1a8993d0
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18394
Signed-off-by: Matt Ranostay <matt.ranostay@intel.com>
Reviewed-on: https://android.intel.com:443/468455
---
 drivers/tty/serial/hsu/hsu_core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/tty/serial/hsu/hsu_core.c b/drivers/tty/serial/hsu/hsu_core.c
index 0783158..3d7c40b 100644
--- a/drivers/tty/serial/hsu/hsu_core.c
+++ b/drivers/tty/serial/hsu/hsu_core.c
@@ -1109,6 +1109,8 @@ serial_hsu_do_set_termios(struct uart_port *port, struct ktermios *termios,
 		cval |= UART_LCR_EPAR;
 
 	baud = uart_get_baud_rate(port, termios, old, 0, 4000000);
+	if (!baud)
+		return;
 
 	if (up->hw_type == hsu_intel) {
 		/*
-- 
1.9.1

